src/iop/toneequal.c

```diff
@@ -1670,31 +1670,21 @@ void gui_update(struct dt_iop_module_t *self)
 
   show_guiding_controls(self);
   invalidate_luminance_cache(self);
+  dt_iop_refresh_preview(self);
 
   dt_bauhaus_widget_set_quad_active(GTK_WIDGET(g->show_luminance_mask), g->mask_display);
 }
 
 void gui_changed(dt_iop_module_t *self, GtkWidget *w, void *previous)
 {
   dt_iop_toneequalizer_gui_data_t *g = (dt_iop_toneequalizer_gui_data_t *)self->gui_data;
-  if(w == g->method     ||
-     w == g->blending   ||
-     w == g->feathering ||
-     w == g->iterations ||
-     w == g->quantization)
-  {
-    invalidate_luminance_cache(self);
-  }
-  else if (w == g->details)
-  {
-    invalidate_luminance_cache(self);
+  if (w == g->details)
     show_guiding_controls(self);
-  }
   else if (w == g->contrast_boost || w == g->exposure_boost)
-  {
-    invalidate_luminance_cache(self);
     dt_bauhaus_widget_set_quad_active(w, FALSE);
-  }
+
+  invalidate_luminance_cache(self);
+  dt_iop_refresh_preview(self);
```

Comment:
why not simply call `dt_iop_refresh_preview` within `invalidate_luminance_cache` ?

Reply:
because I though there might be some cases where it was not needed, but yes, you are right